dialogue-client-test-lib/src/main/java/com/palantir/dialogue/AbstractChannelTest.java

```diff
@@ -196,165 +192,122 @@ public void fillsQueryParameters() throws Exception {
-        assertThatThrownBy(() -> call.execute(observer))
-                .hasMessage("Error, this call was already executed");
-    }
-
-    @Test
-    public void canCancelNonExecutedCalls() {
-        assertThatCode(() -> channel.createCall(endpoint, request).cancel()).doesNotThrowAnyException();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        Futures.addCallback(call, callback, MoreExecutors.directExecutor());
+        verify(callback).onSuccess(any());
+        verify(callback, never()).onFailure(any());
     }
 
     @Test
     public void canCancelMultipleTimes() {
-        Call call = channel.createCall(endpoint, request);
-        call.execute(observer);
-        assertThatCode(call::cancel).doesNotThrowAnyException();
-        assertThatCode(call::cancel).doesNotThrowAnyException();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        assertThatCode(() -> call.cancel(true)).doesNotThrowAnyException();
+        assertThatCode(() -> call.cancel(true)).doesNotThrowAnyException();
     }
 
     @Test
     public void callCancellationIsObservedAsException() throws InterruptedException {
-        channel.createCall(endpoint, request).execute(mock(Observer.class));  // drain enqueued response
+        channel.createCall(endpoint, request);  // drain enqueued response
 
         channel = createChannel(server.url("").url(), Executors.newCachedThreadPool());
-        Call call = channel.createCall(endpoint, request);
-        call.execute(observer);
-        call.cancel();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        Futures.addCallback(call, callback, MoreExecutors.directExecutor());
+        call.cancel(true);
 
         Thread.sleep(1000);
         server.enqueue(new MockResponse());
-        verify(observer, never()).success(any());
+        verify(callback, never()).onSuccess(any());
         ArgumentCaptor<Throwable> throwable = ArgumentCaptor.forClass(Throwable.class);
-        verify(observer).exception(throwable.capture());
+        verify(callback).onFailure(throwable.capture());
         assertThat(throwable.getValue())
                 // Different exceptions for HttpClient vs OkHttpClient
                 .isInstanceOfAny(CancellationException.class, IOException.class)
                 .hasMessageFindingMatch("(Task was cancelled\\.)|(Canceled)");
-        verify(observer, never()).failure(any());
     }
 
     // TODO(rfink): How to test that cancellation propagates to the server?
 
     @Test
     public void connectionErrorsSurfaceAsExceptions() throws IOException {
         server.shutdown();
-        channel.createCall(endpoint, request).execute(observer);
-        verify(observer, never()).success(any());
-        verify(observer).exception(any());
-        verify(observer, never()).failure(any());
-    }
-
-    @Test
-    public void conjureErrorsSurfaceAsFailures() throws IOException {
```

Comment:
what happened to this behavior?

Reply:
Going to have something that maps a `Response` into a `ResponseOrError` sort of thing